client/src/components/Category/index.js

```diff
@@ -0,0 +1,95 @@
+import React, { Component } from "react"
+import './style.css'
+import axios from "axios"
+
+class Category extends Component {
+  state = {
+    details: [],
+    userPledges: [],
+    name: "",
+    categories: [],
+
+  }
+  componentDidMount = () => {
+
+    // request to get all categories
+    const userId = 1
+    axios.get(`/api/dashboard`).then(({ data }) => {
+
+      this.setState({
+        details: data
+      })
+    })
+    // request to get pledges that user make it in all categories
+    axios.get(`/api/dasboard_pledges_in_category/${userId}`).then(({ data }) => {
+      this.setState(
+        { userPledges: data })
+    })
+    // request to get all pledges in each category in general 
+    axios.get(`/api/dashboard_pledges_category`)
+      .then(({ data }) => {
+        this.setState({ categories: data })
+      })
+  }
+  // function to push the data to category page
+  moveToCategorActionPage = (id) => {
+    const { history } = this.props
+
+    history.push({
+      pathname: `action-category/${id}`,
+      data: this.state.details
+    })
+  }
+
+  // function to calculate all pledges that user make it in category by filter
+  numberOfPledges = (arr1, item) => {
+    let arr2 = arr1.filter((element) => {
+      if (element.name === item) {
+        return element.pledge_name
+      }
+    })
+    return arr2.length
+  }
+
+  //function to calculate all pledges in the category by filtering the array
+  numberOfPledges2 = (arr1, item) => {
+    let arr2 = arr1.filter((element) => {
+      if (element.category_id === item) {
+        return element.pledge_id
+      }
+    })
+    return arr2.length
+  }
+
+  render() {
+    const { data } = this.state.details
+    const { userPledges } = this.state.userPledges
+    const { pledgesCategory } = this.state.categories
+    return (
+      <switch>
```

Comment:
Used `Switch` but never defined, you should import it first 
`import {Switch} from "react-router-dom";`
also when you use the tag `<Switch>` use it capitalized

Reply:
done